### PR TITLE
Feat price history

### PR DESCRIPTION
**Pull request by dolcalmi**
- [x] Add db packages
- [x] Add migrations and seeds
- [x] Add price repo
- [x] Add history server
- [x] Refactor realtime impl to avoid db dependency
- [x] Create monorepo structure
- [x] Update github actions + ci to use monorepo
- [ ] Add update history server
- [ ] Add tests

### CODE REVIEW

File: src/servers/history/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,35 @@
+FROM node:16-alpine AS BUILD_IMAGE
```
Comment (bodymindarts) [inline src/servers/history/Dockerfile:1]: Dockerfile should not be under `src` (its not source code).

Either on root level or if there are many images that need to be built in this repo under `image/<type>/Dockerfile`

File: src/servers/history/run.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import healthCheck from "grpc-health-check"
+
+import { History } from "@app"
+import { ServiceStatus } from "@domain/index"
+import { baseLogger } from "@services/logger"
+
+import { protoDescriptorPrice } from "../grpc"
+
+const statusMap = {
+  "": 2,
```
Comment (bodymindarts) [inline src/servers/history/run.ts:11]: This is strange. Empty string?

File: src/servers/realtime/Dockerfile [hunk 3]
```diff
@@ -12,14 +12,14 @@ RUN yarn build
 RUN rm -rf /app/node_modules
 RUN yarn install --frozen-lockfile --production
 
-FROM gcr.io/distroless/nodejs:16
+FROM node:16-alpine
```
Comment (bodymindarts) [inline src/servers/realtime/Dockerfile:15]: should move this docker file too...

Probably should be under `images/{realtime,history}/Dockerfile`

**Review by bodymindarts — COMMENTED** [review]
I consider sharing code between pods quite dangerous (we've had issues with it in main repo in the past).
Why are we putting the code for both of these pods in the same `src` hierarchy?
I think node / yarn support monorepo structures where we have multiple sub-packages in 1 repo. Can we move to that structure to better separate the code for each process?

If we do that then the structure could look something like:
```
realtime
- src
- Dockerfile
history
- src
- Dockerfile
```

File: src/servers/history/run.ts (hunk 2 above)
Comment (dolcalmi) [reply src/servers/history/run.ts:11]: I have not check how works health service but this came from old code https://github.com/GaloyMoney/price/blob/ce60a47c816f20145c725a8c9aeb5e97187e744e/src/index.ts#L14

File: dev/otel-agent-config.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  jaeger:
+    protocols:
+      thrift_compact: # on port 6831
+      thrift_binary: # on port 6832
+
+processors:
+  batch:
+  attributes:
+    actions:
+    - key: apiKey
```
Comment (bodymindarts) [inline dev/otel-agent-config.yaml:11]: If these need redacting you need to PR against https://github.com/GaloyMoney/galoy-infra/blob/main/modules/services/opentelemetry-values.yml as well

**Comment by dolcalmi** [comment]
closed in favor of https://github.com/GaloyMoney/price/pull/19

**Pull request closed without merging**